lighty-modules/lighty-bgp/src/main/java/io/lighty/modules/bgp/deployer/BgpModule.java

```diff
@@ -0,0 +1,237 @@
+        bgpDeployer = new DefaultBgpDeployer(DEFAULT_BGP_NETWORK_INSTANCE_NAME, cssProvider, rpcProvider,
+                ribExtensionConsumerContext, bgpDispatcher, routingPolicyFactory, codecsRegistry, domDataBroker,
+                dataBroker, tableTypeRegistryConsumer, stateCollector);
+        //Topologies
+        bgpTopologyDeployer = new BgpTopologyDeployerImpl(dataBroker, cssProvider);
+        ipv4TopologyProvider = new Ipv4TopologyProvider(bgpTopologyDeployer);
+        ipv6TopologyProvider = new Ipv6TopologyProvider(bgpTopologyDeployer);
+        linkstateTopologyProvider = new LinkstateTopologyProvider(bgpTopologyDeployer);
+        graphServer = new ConnectedGraphServer(dataBroker);
+        linkstateGraphProvider = new LinkstateGraphProvider(bgpTopologyDeployer, graphServer);
+    }
+
+    public BgpModule(final EffectiveModelContext modelContext, final DataBroker dataBroker,
+            final DOMDataBroker domDataBroker, final BindingCodecTreeFactory codecTreeFactory,
+            final BindingRuntimeContext runtimeContext, final RpcProviderService rpcProvider,
+            final ClusterSingletonServiceProvider cssProvider, final BindingNormalizedNodeSerializer serializer,
+            final EventLoopGroup bossGroup, final EventLoopGroup workerGroup) {
+        this(modelContext, dataBroker, domDataBroker, codecTreeFactory.create(runtimeContext),
+                rpcProvider, cssProvider, serializer, bossGroup, workerGroup);
+    }
+
+    public BgpModule(final LightyServices lightyServices) {
+        this(lightyServices.getEffectiveModelContextProvider().getEffectiveModelContext(),
+                lightyServices.getBindingDataBroker(), lightyServices.getClusteredDOMDataBroker(),
+                lightyServices.getBindingCodecTreeFactory(),
+                lightyServices.getAdapterContext().currentSerializer().getRuntimeContext(),
+                lightyServices.getRpcProviderService(), lightyServices.getClusterSingletonServiceProvider(),
+                lightyServices.getBindingNormalizedNodeSerializer(), lightyServices.getBossGroup(),
+                lightyServices.getWorkerGroup());
+    }
+
+
+    @Override
+    @SuppressWarnings({"checkstyle:illegalCatch"})
+    protected boolean initProcedure() {
+        try {
+            //Load initial routing policies into datastore
+            initialConfigLoader.init();
+            simpleStatementRegistry.start();
+            bgpDeployer.init();
+            return true;
+        } catch (Exception e) {
+            LOG.warn("Failed to initialize BGPModule", e);
+            return false;
+        }
+    }
+
+    @Override
+    @SuppressWarnings({"checkstyle:illegalCatch"})
+    protected boolean stopProcedure() {
+        boolean closeSuccess = true;
+        try {
+            simpleStatementRegistry.close();
+        } catch (Exception e) {
+            LOG.warn("Failed to stop BGP statement registry", e);
+            return false;
```

Comment:
closeSuccess should be set to false instead of return false (and not closing other services)